docs/pages/management/guides/joining-nodes-azure.mdx

```diff
@@ -0,0 +1,232 @@
+description: Use the Azure join method to add Nodes to your Teleport cluster on Azure
+---
+
+This guide will explain how to use the **Azure join method** to configure
+Teleport Nodes and Proxy Service instances to join your Teleport cluster without
+sharing any secrets when they are running in an Azure Virtual Machine.
+
+The Azure join method is available in Teleport 12.1+. It is available to any
+Teleport Node or Proxy running in an Azure Virtual Machine.
+
+<Details
+scope={["oss", "enterprise", "cloud"]}
+scopeOnly
+opened
+title="Another Azure Node joining method"
+>
+
+You can also configure Nodes running in Azure to join a cluster via **secret tokens**,
+which is useful when you don't want to rely on Azure-specific APIs.
+Read more in the following guide:
+[Adding Nodes to your Cluster](../admin/adding-nodes.mdx)
+
+</Details>
+
+<Notice type="warning" scope={["oss", "enterprise"]}>
+
+The Azure join method will not work if TLS is terminated at a load balancer in
+front of your Teleport Proxy Service unless the Node using this method is
+connecting directly to the Auth Service.
+
+</Notice>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- An Azure Virtual Machine to act as a Teleport Node, with the Teleport binary
+  installed. The Virtual Machine must have a
+  [Managed Identity](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
+  assigned to it with permission to read virtual machine info.
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/4. Set up a Managed Identity
+Every virtual machine hosting a Node or Proxy using the Azure method to join
+your Teleport cluster needs a Managed Identity assigned to it. The identity 
+requires the `Microsoft.Compute/virtualMachines/read` permission so Teleport
+can look up the virtual machine. No other permissions are required.
+
+### Create a custom role
+
+Enter the name of your Azure resource group in the Azure Portal search box and
+visit the page for that resource group. On the left navigation sidebar, click
+the **Access control (IAM)** tab. In the row of buttons at the top of the
+**Access control (IAM)** panel, click **Add > Add custom role**.
+
+In the **Custom role name** field, enter `teleport-read-vm`.
+
+![Create custom
+role](../../../img/azure/create-custom-role.png)
+
+Click the **Permissions** tab, then within the **Permissions** view, click
+**+Add permissions**.
+
+Enter `Microsoft.Compute/virtualMachines/read` in the search box. Click the
+**Microsoft Compute** box, then enable **Read: Get Virtual Machine**. Click
+**Add**.
+
+![Add virtual machine read
+permission](../../../img/azure/read-permission.png)
+
+Click **Review + create**, then **Create**.
+
+### Create an Azure managed identity
+
+Visit the [Managed
+Identities](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.ManagedIdentity%2FuserAssignedIdentities)
+view in Azure Portal.
+
+Click **Create**.
```

Comment:
These manual instructions are fine, but don't forget about infra as code. A terraform or ARM snippet for creating these resources would be helpful.